### ISSUE TITLE
Rule S2259: False positive on conditional boolean check

### ISSUE THREAD

**Issue by nalla** [SonarSource/sonar-csharp#827]
### Description

In the following code, line 5 leads to a false positive S2259.

~~~C#
public void OnActionExecuted( ActionExecutedContext context )
{
  var descriptor = context.ActionDescriptor as ControllerActionDescriptor;
  if( string.IsNullOrWhiteSpace( descriptor?.ActionName ) ||
      string.IsNullOrWhiteSpace( descriptor.ControllerName ) )
    return;
  [...]
}
~~~

### Expected behavior

No `'descriptor' is null on at least one execution path.` (S2259) should be raised.

### Actual behavior

Rule S2259 is raised.

### Related information

* SonarQube Version 6.3 (build 19869)
* SonarAnalyzer for C# version 5.9.0.1001
* SonarQube Runner Version 2.4

**Comment by Evangelink** [comment]
Hi @nalla ,
Sorry for this late answer but I am not able to reproduce the issue with the following snippet on the last version of SonarC#:
```
class A
{
    public string Foo;
    public string Bar;
}

class B
{
    public void Do(A a)
    {
        if (string.IsNullOrWhiteSpace(a?.Foo) || string.IsNullOrWhiteSpace(a.Bar)) {}
    }
}
```
Could you try to update your SonarC# version and let us know if you still encounter the issue?

**Comment by nalla** [comment]
We already issued an upgrade of our instance. Will take some time. We will try and let you know. Thanks!

**Comment by Evangelink** [comment]
May I close the ticket and let you reopen it (or ping me) when you have made more tests?

**Comment by nalla** [comment]
Sure. I can even close it for you ;)

**Comment by nalla** [comment]
@Evangelink just a quick update, the installation of the latest update resolved the issue.

**Comment by Evangelink** [comment]
@nalla Thx for update!
